### PR TITLE
qa/crontab: update crontab for nightlies to include squid (+)

### PR DESCRIPTION
**Pull request by batrick**
This is a fairly big overhaul of nightlies. We're bringing back weekly testing of the main and release branches. Squid pre-release will have suites tested twice a week and at higher priority (-p 100) which is the highest QE priority [1].

[1] https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-intro/#testing-priority

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by batrick** [comment]
@yuriw I forget, do we test these crontab changes somewhere before hand or just "test it in production"?

**Comment by batrick** [comment]
Adding leads.

File: qa/crontab/teuthology-cronjobs [hunk 1]
```diff
@@ -31,127 +31,100 @@ CEPH_QA_EMAIL="ceph-qa@ceph.io"
...
-
-25 01 * * 4 CEPH_BRANCH=reef; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH -n 100 -m $MACHINE_NAME -s upgrade-clients/client-upgrade-quincy-reef -k distro -e $CEPH_QA_EMAIL --suite-branch quincy --suite-repo https://github.com/ceph/ceph.git --ceph-repo https://github.com/ceph/ceph.git -p 50 --force-priority
+20 01 * * 3 CEPH_BRANCH=reef; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH -n 100 -m $MACHINE_NAME -s upgrade:pacific-x -k distro -e $CEPH_QA_EMAIL -p 170
+25 01 * * 3 CEPH_BRANCH=reef; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH -n 100 -m $MACHINE_NAME -s upgrade:quincy-x -k distro -e $CEPH_QA_EMAIL -p 170
+30 01 * * 3 CEPH_BRANCH=reef; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH -n 100 -m $MACHINE_NAME -s upgrade-clients/client-upgrade-octopus-reef -k distro -e $CEPH_QA_EMAIL --suite-branch octopus --suite-repo https://github.com/ceph/ceph.git --ceph-repo https://github.com/ceph/ceph.git -p 170
```
Comment (adk3798) [inline qa/crontab/teuthology-cronjobs:118]: is this test actually doing upgrades from octopus to reef?

**Review by adk3798 — APPROVED** [review]
Can't speak to the actual crontab changes, but I approve of the idea of the PR.

File: qa/crontab/teuthology-cronjobs [hunk 2]
```diff
@@ -31,127 +31,100 @@ CEPH_QA_EMAIL="ceph-qa@ceph.io"
...
-## The suites below run without filters
-
-# 47 01 * * 5  CEPH_BRANCH=octopus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 100 -m $MACHINE_NAME -s upgrade/client-upgrade-luminous-octopus -k distro -e $CEPH_QA_EMAIL --suite-branch luminous -t py2
-# 50 01 * * 5  CEPH_BRANCH=octopus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 100 -m $MACHINE_NAME -s upgrade/client-upgrade-mimic-octopus -k distro -e $CEPH_QA_EMAIL --suite-branch mimic -t py2
-# 50 01 * * 5  CEPH_BRANCH=octopus; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH  -n 100 -m $MACHINE_NAME -s upgrade-clients/client-upgrade-nautilus-octopus -k distro -e $CEPH_QA_EMAIL --suite-branch nautilus
+# rados is massive and difficult to bring down to less than 300 jobs, use one higher priority
+00 05 * * 0   CEPH_BRANCH=main; MACHINE_NAME=smithi; SUITE_NAME=rados;         KERNEL=distro;  /home/teuthology/bin/cron_wrapper /home/teuthology/bin/schedule_subset.sh 100000 $CEPH_BRANCH $MACHINE_NAME $SUITE_NAME $CEPH_QA_EMAIL $KERNEL -p 201
```
Comment (idryomov) [inline qa/crontab/teuthology-cronjobs:58]: Currently Yuri is scheduling runs on main by hand and sending out `main baseline $DATE` emails, also on a weekly basis.  Is this supposed to replace that?

File: qa/crontab/teuthology-cronjobs [hunk 3]
```diff
@@ -31,127 +31,100 @@ CEPH_QA_EMAIL="ceph-qa@ceph.io"
...
 ## ********** teuthology/nop on main branch - daily
 0 0 * * * CEPH_BRANCH=main; MACHINE_NAME=smithi; /home/teuthology/bin/cron_wrapper teuthology-suite -v -c $CEPH_BRANCH -n 10 -m $MACHINE_NAME -s teuthology/nop -k distro -e $CEPH_QA_EMAIL -p 50
 
-## quincy branch runs - weekly
+
+## main branch runs - weekly
 ## suites rados and rbd use --subset arg and must be call with schedule_subset.sh
 ## see script in https://github.com/ceph/ceph/tree/main/qa/machine_types
 
-# 01 07 * * 0 CEPH_BRANCH=quincy; MACHINE_NAME=smithi; SUITE_NAME=rados;         KERNEL=distro;  /home/teuthology/bin/cron_wrapper /home/teuthology/bin/schedule_subset.sh 100000 $CEPH_BRANCH $MACHINE_NAME $SUITE_NAME $CEPH_QA_EMAIL $KERNEL
```
Comment (idryomov) [inline qa/crontab/teuthology-cronjobs:53]: Runs on quincy seem to have disappeared, even commented out.  It doesn't look like that was intended given
>  We're bringing back weekly testing of the main and release branches.

description.

File: qa/crontab/teuthology-cronjobs (hunk 2 above)
Comment (batrick) [reply qa/crontab/teuthology-cronjobs:58]: Yes.

File: qa/crontab/teuthology-cronjobs (hunk 3 above)
Comment (batrick) [reply qa/crontab/teuthology-cronjobs:53]: Quincy is about to be EOL so I thought bringing back nightlies for it was not necessary. The upgrade nightlies were not commented so I kept them.

File: qa/crontab/teuthology-cronjobs (hunk 1 above)
Comment (batrick) [reply qa/crontab/teuthology-cronjobs:118]: This is a suite that's been added to the release branches by @yuriw in the past. It does not exist in `reef`. I'm removing it from the crontab for now.

File: qa/crontab/teuthology-cronjobs [hunk 4]
```diff
@@ -31,127 +31,97 @@ CEPH_QA_EMAIL="ceph-qa@ceph.io"
...
+30 05 * * 3   CEPH_BRANCH=main; MACHINE_NAME=smithi; SUITE_NAME=fs;            KERNEL=distro;  /home/teuthology/bin/cron_wrapper /home/teuthology/bin/schedule_subset.sh    512 $CEPH_BRANCH $MACHINE_NAME $SUITE_NAME $CEPH_QA_EMAIL $KERNEL -p 200
+40 05 * * 4   CEPH_BRANCH=main; MACHINE_NAME=smithi; SUITE_NAME=powercycle;    KERNEL=distro;  /home/teuthology/bin/cron_wrapper /home/teuthology/bin/schedule_subset.sh      4 $CEPH_BRANCH $MACHINE_NAME $SUITE_NAME $CEPH_QA_EMAIL $KERNEL -p 200
+50 05 * * 5   CEPH_BRANCH=main; MACHINE_NAME=smithi; SUITE_NAME=rgw;           KERNEL=distro;  /home/teuthology/bin/cron_wrapper /home/teuthology/bin/schedule_subset.sh      1 $CEPH_BRANCH $MACHINE_NAME $SUITE_NAME $CEPH_QA_EMAIL $KERNEL -p 200
+60 05 * * 6   CEPH_BRANCH=main; MACHINE_NAME=smithi; SUITE_NAME=krbd;          KERNEL=testing; /home/teuthology/bin/cron_wrapper /home/teuthology/bin/schedule_subset.sh      1 $CEPH_BRANCH $MACHINE_NAME $SUITE_NAME $CEPH_QA_EMAIL $KERNEL -p 200
```
[the remark this answers is not in the document]
Comment (batrick) [reply qa/crontab/teuthology-cronjobs:64]: ACK

**Review by idryomov — APPROVED** [review]
Ack from the RBD perspective

**Comment by batrick** [comment]
jenkins render docs

**Review by rzarzynski — APPROVED** [review]
ACK from RADOS.

**Review by ljflores — APPROVED** [review]

**Pull request merged**